dimod/decorators.py

```diff
@@ -188,7 +188,7 @@ def structured_sample(sampler, bqm, **kwargs):
 
         if not adjacency.keys() >= bqm.variables:
             raise BinaryQuadraticModelStructureError(
-                f"given bqm contains a variable, {(adjacency.keys() - bqm.variables).pop()!r}, "
+                f"given bqm contains a variable, {(list(bqm.variables) - adjacency.keys())!r}, "
```

Comment:
I think you still want  `(list(bqm.variables) - adjacency.keys()).pop()` since the error message says "variable". Otherwise you want to say "variables".

Reply:
I think more than one variable is likely to be more common, so I'd go with variables

Reply:
I think that's true, but it's also less likely to be readable. Since it's likely to be quite a long list, potentially thousands. Whereas "there is at least one variable, x" provides enough info and is very easy to read.

Reply:
Another option is to truncate the set repr to a fixed string length...

Reply:
:biking_man: :house: 

:laughing: